api/python/t4/packages.py

```diff
@@ -195,6 +245,28 @@ def __contains__(self, logical_key):
         """
         return logical_key in self._data
 
+    def __getitem__(self, prefix):
+        """
+        Filters the package based on prefix, and returns either a new Package
+            or a PackageEntry.
+
+        Args:
+            prefix(str): prefix to filter on
+
+        Returns:
+            PackageEntry if prefix matches a logical_key exactly
+            otherwise Package
+        """
+        if prefix in self._data:
+            return self._data[prefix]
+        result = Package()
+        slash_prefix = prefix.strip('/') + '/' # ensure it ends with exactly one /
```

Comment:
1) Should it be the first line in the function? Otherwise, if `foo.txt` is a key, then `foo.txt` and `foo.txt/` give different results.
2) Should it be `rstrip` (at least, that's what the comment says)?